### ISSUE TITLE
select_one field behavior in Observations panel

### ISSUE THREAD

**Issue by okdistribute** [digidem/mapeo-desktop#344]
**Describe the bug**
Here is a video of the confusing behavior
https://drive.google.com/file/d/17GXr2YpomcgTGdsoysDGDQPvXazUuA8j/view

**Desktop:**
 - Version 5.2.x

**Comment by okdistribute** [comment]
This seems to also be present in 5.1 and below, so I think this is mostly an enhancement request on current behavior rather than a bug.

Current behavior indicates that as you type, field values will be filtered based upon your input text. But this also means you could type other values that aren't options in the mobile app, leading to potentially different values that aren't specified as valid options in the settings.
	- Is this on purpose?

Because of this typing behavior, it also means when you click on a "select one" field, you will only see options that contain the same text, which will give users the behavior jen mentioned seemed like a bug. However, this seems to be made deliberate in the code. In react-mapfilter, we are using a custom-built select handler instead of using material-ui's `select` behavior. 
Check it out here: https://material-ui.com/components/text-fields/#select
	- Do we want to have the same behavior as this component? This would be much easier to maintain since we wouldn't have to maintain our own custom code here.
However, it would mean that users couldn't type custom answers for these fields though. 

Can someone help me understand the desired behavior of this component? @jencastrodoesstuff @aliya-ryan @gmaclennan

**Comment by gmaclennan** [comment]
Hi @karissa, the select_one component was implemented as an auto-complete component following the [integration examples in material-ui](https://v4-5-2.material-ui.com/components/integrated-autocomplete/). The reason I implemented it as an auto-complete rather than select-one is that on Desktop I understood that there was often a need to enter a value that is not included in the selection list. This need might have changed or I might have misunderstood it, and it might be ok to leave it as a restricted list of options.

The latest material-ui release includes a [new autocomplete component](https://material-ui.com/components/autocomplete/) in their "Lab" that I was not aware of when I implemented this (I think it is new). This is almost certainly better tested than our implementation and would be worth switching to if we desire the auto-complete functionality.

The current implementation was intended to show a list of all options when the user focusses the input (e.g. no matter what is in the input, all options are shown when you click or tab into the input) and to then filter the list as the use types. I don't think it is working entirely as intended - forms are notoriously hard to get right.

I'd be interested to hear from @jencastrodoesstuff and @aliya-ryan:

1. Is it important to have the option to enter an "other" value for single-select and/or multi-select fields? (e.g. enter a value that is not in the list of options for the field).
2. Is it important to configure when it is possible to enter an "other" value, e.g. should it be possible for all single/multi-select fields, or only for some of them when it is enabled in the preset file?
3. Is an auto-complete input a good UX for this? Some of the available display options for the Material UI auto-complete can be browsed here: https://material-ui.com/components/autocomplete/
